### ISSUE TITLE
Yolov4 head sub_module fails in main

### ISSUE THREAD

**Issue by punithsekar** [tenstorrent/tt-metal#11998]
**Describe the bug**
yolov4 head sub_module fails in main after  [this commit](https://github.com/tenstorrent/tt-metal/commit/7cd61f48619c12fd655ec1d046e438df3aa29069) in main.

```
RuntimeError: TT_FATAL @ ../ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp:977: bias_shape.without_padding()[-1] == b_shape[-1]
E       info:
E       Unsupported bias shape: last dimension not equal to second input's last dimension.
E       backtrace:
E        --- ttnn::operations::matmul::Matmul::validate(std::__1::vector<tt::tt_metal::Tensor, std::__1::allocator<tt::tt_metal::Tensor>> const&, std::__1::vector<std::__1::optional<tt::tt_metal::Tensor const>, std::__1::allocator<std::__1::optional<tt::tt_metal::Tensor const>>> const&) const 
```

**To Reproduce**

1. Checkout [keerthanar/yolov4_bias_issue_unit_test](https://github.com/tenstorrent/tt-metal/tree/keerthanar/yolov4_bias_issue_unit_test)
2. Run `pytest tt-metal/tests/ttnn/unit_tests/operations/test_new_conv2d.py::test_yolov4_bias_issue_case`

**Please complete the following environment information:**
- OS: Ubuntu 20.04
- Device- n150

**Comment by punithsekar** [comment]
Temporarily, I have skipped those test in main.

PR #11959

**Comment by mywoodstock** [comment]
cc: @bbradelTT

**Comment by keerthana-r-mcw** [comment]
Hi @sankarmanoj-tt , Even after pulling the latest changes from the main branch, the issue persists.

**Comment by bbradelTT** [comment]
Starting to work on this.

1. I think test_yolov4_bias_issue_case is duplicated in the file on the branch
2. I think the repro step should be
```
pytest tests/ttnn/unit_tests/operations/test_new_conv2d.py::test_yolov4_bias_issue_case
```
3. I can check for both padded and non-padded. That allows the test to pass. I need to see if there are any negative consequences. If not, I'll get a PR out.

**Comment by keerthana-r-mcw** [comment]
Hi @bbradelTT , The issue got resolved with this [matmul_op.cpp](https://github.com/tenstorrent/tt-metal/commit/bd6eb6cf3ca9389479fce163686ffdf86f889b45#diff-7b5f8d16e3f4117e06d1c4a13d39596f7f25ac5a023dfdb386fe1c3674d3bbd9) fix.

**Comment by punithsekar** [comment]
The issue has been solved.
#12503 PR for removing the skip statements.

**Comment by punithsekar** [comment]
Merged #12503 to main.
Closing this issue.
